src/languageTools/LanguageTools.js

```diff
@@ -0,0 +1,366 @@
+        this._notify(ToolingInfo.SYNCHRONIZE_EVENTS.DOCUMENT_CLOSED, params);
+    };
+
+    //didChangeTextDocument
+    LanguageClientProxy.prototype.notifyTextDocumentChanged = function (params) {
+        this._notify(ToolingInfo.SYNCHRONIZE_EVENTS.DOCUMENT_CHANGED, params);
+    };
+
+    //didSaveTextDocument
+    LanguageClientProxy.prototype.notifyTextDocumentSave = function (params) {
+        this._notify(ToolingInfo.SYNCHRONIZE_EVENTS.DOCUMENT_SAVED, params);
+    };
+
+    /**
+        Custom messages
+     */
+
+    //customNotification
+    LanguageClientProxy.prototype.sendCustomNotification = function (params) {
+        this._notifyClient(ToolingInfo.LANGUAGE_SERVICE.CUSTOM_NOTIFICATION, params);
+    };
+
+    //customRequest
+    LanguageClientProxy.prototype.customRequest = function (params) {
+        return this._requestClient(ToolingInfo.LANGUAGE_SERVICE.CUSTOM_REQUEST, params);
+    };
+
+    //TODO: this needs to be moved to tooling manager
+    LanguageClientProxy.prototype._handleActiveEditorChange = function (event, current, previous) {
+        var self = this;
+
+        function _sendDocumentOpenNotification(languageId, document) {
+            if (self._languages.includes(languageId)) {
+                self.notifyTextDocumentOpened({
+                    languageId: languageId,
+                    filePath: (document.file._path || document.file.fullPath),
+                    fileContent: document.getText()
+                });
+            }
+        }
+
+        if (previous) {
+            previous.document
+                .off("languageChanged.language-tools");
+            self.notifyTextDocumentClosed({
+                filePath: (previous.document.file._path || previous.document.file.fullPath)
+            });
+        }
+        if (current) {
+            var currentLanguageId = LanguageManager.getLanguageForPath(current.document.file.fullPath).getId();
+            current.document
+                .on("languageChanged.language-tools", function () {
+                    //TODO: Also need to attach listeners for Parameter Hints
+                    _sendDocumentOpenNotification(currentLanguageId, current.document);
+                });
+            //TODO: Also need to attach listeners for Parameter Hints
+            _sendDocumentOpenNotification(currentLanguageId, current.document);
+        }
+    };
+
+    //TODO: this needs to be moved to tooling manager
+    LanguageClientProxy.prototype._handleProjectOpen = function () {
+        var self = this;
+
+        self.stop().done(function() {
+            self.start();
+        });
+        //TODO : the case where server supports change of workspace folders
+        //TODO : Reload with extensions case
+    };
+
+    function _attachEventHandlers(client) {
+        EditorManager.on("activeEditorChange.language-tools", client._handleActiveEditorChange.bind(client));
+        ProjectManager.on("projectOpen", client._handleProjectOpen.bind(client));
```

Comment:
This will work only if file or Project is opened after extension is loaded.
Need a check here that if there is any editor is active. _handleActiveEditor should be called and then attach event handlers.